File: polyglot/runtime_template_resolver/mjs/src/sdk.ts
import { TemplateResolver } from './resolver.js';
import { IResolverOptions, ITemplateResolver } from './interfaces.js';
import { validatePlaceholder } from './validator.js';
import { extractPlaceholders } from './extractor.js';
import { compile as compileTemplate } from './compiler.js';

const defaultResolver = new TemplateResolver();

export const SDK = {
    resolve(template: string, context: Record<string, unknown>, options?: IResolverOptions): string {
        return defaultResolver.resolve(template, context, options);
    },
    resolveMany(templates: string[], context: Record<string, unknown>, options?: IResolverOptions): string[] {
        return templates.map(t => defaultResolver.resolve(t, context, options));
    },
    resolveObject(obj: unknown, context: Record<string, unknown>, options?: IResolverOptions): unknown {
        return defaultResolver.resolveObject(obj, context, options);
    },
    validate(template: string): void {
        const placeholders = extractPlaceholders(template);
        // split pipe for default values: key | "default"
        placeholders.forEach(p => {
            // Basic extraction of key part
            let key = p;
            if (p.includes('|')) {
                key = p.split('|')[0];
            }
            validatePlaceholder(key.trim());
        });
    },
    extract(template: string): string[] {
        return extractPlaceholders(template);
    },
    compile(template: string) {
        return compileTemplate(template);
    }
};

File: polyglot/runtime_template_resolver/py/src/runtime_template_resolver/sdk.py
from typing import List, Any, Dict, Optional
from .resolver import TemplateResolver
from .interfaces import ResolverOptions
from .validator import validate_placeholder
from .extractor import extract_placeholders
from .compiler import compile as compile_template

_default_resolver = TemplateResolver()

def resolve(template: str, context: Dict[str, Any], options: Optional[ResolverOptions] = None) -> str:
    return _default_resolver.resolve(template, context, options)

def resolve_many(templates: List[str], context: Dict[str, Any], options: Optional[ResolverOptions] = None) -> List[str]:
    return [_default_resolver.resolve(t, context, options) for t in templates]

def resolve_object(obj: Any, context: Dict[str, Any], options: Optional[ResolverOptions] = None) -> Any:
    return _default_resolver.resolve_object(obj, context, options)

def validate(template: str) -> None:
    placeholders = extract_placeholders(template)
    for p in placeholders:
        key = p
        if "|" in p:
            key = p.split("|")[0]
        validate_placeholder(key.strip())

def extract(template: str) -> List[str]:
    return extract_placeholders(template)

def compile(template: str):
    return compile_template(template)
